### ISSUE TITLE
What does each index of [:,1:,:,:,:] mean？

### ISSUE THREAD

**Issue by zjx103-orange** [jocpae/clDice#6]
Can you explain what each index represents? My task is a segmentation task of two classifications (foreground and background), and the output of the network is 4-dimensional (batchsize, 2, H, W). I want to try the loss function proposed in the paper, but I haven't figured out why it is 5-dimensional.

**Comment by jocpae** [comment]
Hi, thank you for your comment, we have changed the notation to `[:,1:,...]`, as all the dimensions after the foreground denote the spatial dimensions. (I asuume your case involves 2D data, making the 5th dimension redundant.) On why we only use the foreground please refer to our paper and a previous issue.

**Comment by innat** [comment]
@jocpae 
I've just looked into the keras implementation and seems like it has some critical issue.

```python
pres = (K.sum(tf.math.multiply(skel_pred, y_true)[:,1:,:,:,:])  
rec = (K.sum(tf.math.multiply(skel_true, y_pred)[:,1:,:,:,:])
``` 

This is for `channel_first` cases, i.e. `batch_size, channel, w, h, d`. But in soft erode or dilation, the `data_format` is set to `None`, which means `channel_last`. 

```python
p1 = -KL.MaxPool3D(pool_size=(3, 3, 1), strides=(1, 1, 1), padding='same', data_format=None)(-img)
p2 = -KL.MaxPool3D(pool_size=(3, 1, 3), strides=(1, 1, 1), padding='same', data_format=None)(-img)
p3 = -KL.MaxPool3D(pool_size=(1, 3, 3), strides=(1, 1, 1), padding='same', data_format=None)(-img)
```

**Comment by innat** [comment]
Another critical issue

currently 

```python
p1 = -KL.MaxPool3D(pool_size=(3, 3, 1), strides=(1, 1, 1), padding='same', data_format=None)(-img)
p2 = -KL.MaxPool3D(pool_size=(3, 1, 3), strides=(1, 1, 1), padding='same', data_format=None)(-img)
p3 = -KL.MaxPool3D(pool_size=(1, 3, 3), strides=(1, 1, 1), padding='same', data_format=None)(-img)
```

but should be 

```python
p1 = -KL.MaxPool3D(pool_size=(3, 1, 1), strides=(1, 1, 1), padding='same', data_format=None)(-img)
p2 = -KL.MaxPool3D(pool_size=(1 3, 1), strides=(1, 1, 1), padding='same', data_format=None)(-img)
p3 = -KL.MaxPool3D(pool_size=(1, 1, 3), strides=(1, 1, 1), padding='same', data_format=None)(-img)
```
